middleware/legacy.js

```diff
@@ -1,31 +1,46 @@
 // This middleware is responsible for handling redirects for legacy URLs used on
 // previous versions of the Europeana Portal.
 
-function stringifyPathMatch(regexpMatch) {
-  return regexpMatch.slice(1).filter((substring) => typeof substring !== undefined).join('');
+function stringifyPathChunks(chunks) {
+  return chunks.filter((chunk) => typeof chunk !== undefined).join('');
 }
 
-export default ({ redirect, route }) => {
+const rules = [
   // Remove legacy /portal prefix
-  const legacyPortalPrefixPattern = /^\/portal(\/.*)$/;
-  const legacyPortalPrefixMatch = route.path.match(legacyPortalPrefixPattern);
-  if (legacyPortalPrefixMatch) {
-    return redirect({
-      path: legacyPortalPrefixMatch[1],
-      query: route.query
-    });
-  }
-
+  (routePath) => {
+    const legacyPortalPrefixPattern = /^\/portal(\/.*)$/;
+    const legacyPortalPrefixMatch = routePath.match(legacyPortalPrefixPattern);
+    return legacyPortalPrefixMatch ? legacyPortalPrefixMatch[1] : null;
+  },
   // Remove .html from record page URLs
-  const legacyRecordPagePattern = /^(\/[a-z]{2})?(\/record\/[0-9]+\/[a-zA-Z0-9_]+)\.html$/;
-  const legacyRecordPageMatch = route.path.match(legacyRecordPagePattern);
-  if (legacyRecordPageMatch) {
-    const redirectPath = stringifyPathMatch(legacyRecordPageMatch);
-    return redirect({
-      path: redirectPath,
-      query: route.query
-    });
+  (routePath) => {
+    const legacyRecordPagePattern = /^(\/[a-z]{2})?(\/record\/[0-9]+\/[a-zA-Z0-9_]+)\.html$/;
+    const legacyRecordPageMatch = routePath.match(legacyRecordPagePattern);
+    return legacyRecordPageMatch ? stringifyPathChunks(legacyRecordPageMatch.slice(1)) : null;
+  },
```

Comment:
why not have this remove '.html' from the end of _any_ path? 
```const legacyHTMLPattern = /^(.*)?.html$/;```

In conjunction with the removal of /portal, record pages will then be in the right format, and as all the checks are ran anyway this will simplify the path for future checks.